skbio/tree/tests/test_tree.py

```diff
@@ -598,8 +600,22 @@ def test_tip_tip_distances_non_tip_endpoints(self):
 
     def test_tip_tip_distances_no_length(self):
         t = TreeNode.read(StringIO(u"((a,b)c,(d,e)f);"))
-        with self.assertRaises(NoLengthError):
-            t.tip_tip_distances()
+        with warnings.catch_warnings(record=True):
+            warnings.simplefilter("error")
+            with self.assertRaises(RepresentationWarning):
+                t.tip_tip_distances()
```

Comment:
Interesting! I recommend using `numpy.testing.assert_warns`, which simultaneously asserts that the warning is raised and allows a return value to be passed through. See `skbio/io/tests/test_base.py` for an example of how this is used. This also avoids having to use `warnings` to catch and filter warnings.